README.md

```diff
@@ -31,3 +31,35 @@ Once you've imported our library, make sure your API key is available in the HYP
 That's it! Open the Hyperdash mobile app and you should see your logs start to stream in!
 
 Don't have an API key? Run `hyperdash signup` to get one!
+
+### API key storage
+
+If you signed up through the CLI, then your API key is already installed in hyperdash.json file in the home directory of your user.
+
+You can alternatively override this API key with a hyperdash.json file in your local directory (so you can have different API keys for different projects) or with the HYPERDASH_API_KEY environment variable.
+
+Finally, the monitor function accepts an api_key_getter argument that if passed in will be called everytime we need to retrieve your API key. Example:
+
+```
+# test_script.py
+
+from hyperdash.sdk import monitor
+
+# This function can do anything you want, as long as it returns a Hyperdash API key as a string
+def get_api_key():
+  return "super_secret_api_key"
+
+@monitor("dogs vs. cats", api_key_getter=get_api_key)
```

Comment:
@andrewschreiber The only issue with not allowing them to define a custom function is that they can't implement rotation without doing a lot of work on their own.

You might be right that is over-complicated though, I'll make a separate issue for it and we can decide, don't want to strip it out in this diff.